modules/gdscript/language_server/gdscript_language_protocol.cpp

```diff
@@ -189,14 +191,15 @@ void GDScriptLanguageProtocol::poll() {
 	if (server->is_connection_available()) {
 		on_client_connected();
 	}
-	if (peer != NULL) {
+	for(int c_ind=0; c_ind < clients.size(); c_ind++) {
+		Ref<StreamPeerTCP> peer = clients.get(c_ind);
 		StreamPeerTCP::Status status = peer->get_status();
 		if (status == StreamPeerTCP::STATUS_NONE || status == StreamPeerTCP::STATUS_ERROR) {
-			on_client_disconnected();
+			on_client_disconnected(c_ind);
```

Comment:
Is this safe? `on_client_disconnect` modifies `clients`, which you're currently iterating over. I think that's undefined behavior

Reply:
yeah you're right, lemme fix that

Reply:
I tried to make it cause a problem but it refused lol. Undefined behavior is my least favorite part of C++

Reply:
I'd rather write rust, but hey what can you do.